include/dao/db_migrate.js

```diff
@@ -0,0 +1,138 @@
+var url      = require('url');
+var async    = require('async');
+var util     = require('../util.js');
+
+module.exports = function DBMigrateModule(pb) {
+    /**
+     * Array of collections used to append a "site" value to all documents
+     * @private
+     * @static
+     * @readonly
+     * @property MIGRATE_ALL
+     * @type {string[]}
+     */
+    var MIGRATE_ALL = [
+        'article',
+        'comment',
+        'custom_object_type',
+        'custom_object',
+        'media',
+        'page',
+        'plugin',
+        'plugin_settings',
+        'section',
+        'theme_settings',
+        'topic'
+    ];
+
+    var SITE_SPECIFIC_USERS = [
+        pb.security.ACCESS_EDITOR,
+        pb.security.ACCESS_WRITER,
+        pb.security.ACCESS_USER
+    ];
+
+    var SITE_SPECIFIC_SETTINGS = [
+        'active_theme',
+        'content_settings',
+        'section_map',
+        'email_settings'
+    ];
+
+    function DBMigrate() {
+
+        this.run = function (cb) {
+            var self = this;
+            var siteService = new pb.SiteService();
+            siteService.getSiteMap(function (err, result) {
+                if (pb.config.multisite && result.active.length === 0 && result.inactive.length === 0) {
+                    self.createSite(function (err, isTaken, field, result) {
+                        self.siteUid = result.uid;
+                        var tasks = [
+                            util.wrapTask(self, self.migrateContentAndPluginData),
+                            util.wrapTask(self, self.migrateSettings),
+                            util.wrapTask(self, self.migrateUsers)
+                        ];
+                        //self.migrateContentAndPluginData(cb);
```

Comment:
No longer needed.  It has been removed.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Sunday, July 12, 2015 at 7:15 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In include/dao/db_migrate.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34426893>:

> +
> +    function DBMigrate() {
> +
> +        this.run = function (cb) {
> +            var self = this;
> +            var siteService = new pb.SiteService();
> +            siteService.getSiteMap(function (err, result) {
> +                if (pb.config.multisite && result.active.length === 0 && result.inactive.length === 0) {
> +                    self.createSite(function (err, isTaken, field, result) {
> +                        self.siteUid = result.uid;
> +                        var tasks = [
> +                            util.wrapTask(self, self.migrateContentAndPluginData),
> +                            util.wrapTask(self, self.migrateSettings),
> +                            util.wrapTask(self, self.migrateUsers)
> +                        ];
> +                        //self.migrateContentAndPluginData(cb);


Is the commented code needed?

-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34426893>.